src/sentry/sentry_metrics/indexer/base.py

```diff
@@ -0,0 +1,44 @@
+from enum import Enum
+from typing import Optional
+
+from sentry.models import Project
+from sentry.utils.services import Service
+
+
+class UseCase(Enum):
+    METRIC = 0
+    TAG_KEY = 1
+    TAG_VALUE = 2
+
+
+class StringIndexer(Service):  # type: ignore
+    """
+    Provides integer IDs for metric names, tag keys and tag values
+    and the corresponding reverse lookup.
+    """
+
+    __all__ = ("record", "resolve", "reverse_resolve")
+
+    def record(self, project: Project, use_case: UseCase, string: str) -> int:
```

Comment:
I wonder if this should be defined by project or by organization. 
Imagine a metric id. If we partition by project, the same metric (string) will have different numeric ids in different projects. If this is fine, then partitioning by project id is good. Can there be any case where this is not true ?

Reply:
Agreed, we have this case in Release Health even. Updated.